### PR TITLE
 Use artifact downloader hook in fetching markdown html

### PR DESCRIPTION
**Pull request by pnaik1**
https://issues.redhat.com/browse/RHOAIENG-9738

## Description
Updated the `useArtifactStorage` function to make it more stable
Updated the `manifests/common/crd/odhdashboardconfigs.opendatahub.io.crd.yaml`
Updated the compare run section to use the new hook `useArtifactStorage`
Updated the artifact nodes virtualization side drawer panel to use new hook `useArtifactStorage`

## How Has This Been Tested?
Update the crd ,check if disableArtifactsAPI is false the fetch is from the DSP server
when disableArtifactsAPI is true and disableS3Endpoint is false check the fetching occurs from backend 
while running locally 
`open -n -a /Applications/Google\ Chrome.app/Contents/MacOS/Google\ Chrome --args --user-data-dir="/tmp/chrome_dev_test" --disable-web-security` to avoid cors error
1. Upload a pipeline with metrics 
2. Create a run with uploaded pipeline
3. Go to compare runs and check in the markdown section, check the behaviour
4.Go to the created run page and click on the node, from the right side panel, click on the visualization tab, check the behaviour

## Test Impact
N/A

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

File: frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/artifacts/ArtifactVisualization.tsx [hunk 1]
```diff
@@ -61,23 +58,25 @@ export const ArtifactVisualization: React.FC<ArtifactVisualizationProps> = ({ ar
...
-            await fetchStorageObject(namespace, path)
+            await artifactStorage
+              .getStorageObject(artifact)
               .then((text) => setDownloadedArtifact(text))
               .catch(() => null);
             setLoading(false);
           };
           setLoading(true);
-          setDownloadedArtifact(null);
+          setLoading(true);
```
[the remark this answers is not in the document]
Comment (pnaik1) [reply frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/artifacts/ArtifactVisualization.tsx:73]: nice catch 🫣, I will remove this

**Comment by pnaik1** [comment]
The PR is been blocked due to browser running into `CORS` issues on s3 pre-signed URL

**Comment by pnaik1** [comment]
added cypress test for compare runs,

**Review by DaoDaoNoCode — APPROVED** [review]
Tested on a live cluster, lgtm
/lgtm
Screenshot 2024-08-12 at 3 39 49 PM
Screenshot 2024-08-12 at 3 39 40 PM
Screenshot 2024-08-12 at 3 39 28 PM

**Comment by Gkrumbach07** [comment]
This looks good, but why is the markdown not rendering anymore, html looks fine, but markdown is rendering as plain text @pnaik1

**Comment by pnaik1** [comment]
@Gkrumbach07 , I see iframe renders same way the browser does 
I double checked with the artifactlink which opens the new tab
its same

Screenshot 2024-08-13 at 11 24 24 AM

**Pull request merged**